rdt/transformers/pii/anonymizer.py

```diff
@@ -209,3 +214,139 @@ def __repr__(self):
+    ``Faker`` provider and ``function``.
+
+    Args:
+        provider_name (str):
+            The name of the provider in ``Faker``. If ``None`` the ``BaseProvider`` is used.
+            Defaults to ``None``.
+        function_name (str):
+            The name of the function to use within the ``faker.provider``. Defaults to
+            ``lexify``.
+        function_kwargs (dict):
+            Keyword args to pass into the ``function_name`` when being called.
+        locales (list):
+            List of localized providers to use instead of the global provider.
+    """
+
+    MAX_ATTEMPTS_FOR_UNIQUE_VALUES = 10
+    OUTPUT_SDTYPES = {'value': 'categorical'}
+    NEXT_TRANSFORMER = {
+        'value': LabelEncoder(add_noise=True)
+    }
+
+    def __getstate__(self):
+        """Return a dictionary representation of the instance and warn the user when pickling."""
+        LOGGER.warning((
+            'You are saving the mapping information, which includes the original data. '
+            'Sharing this object with others will also give them access to the original data '
+            'used with this transformer.'
+        ))
+
+        return self.__dict__
+
+    def __init__(self, provider_name=None, function_name=None, function_kwargs=None, locales=None):
+        super().__init__(
+            provider_name=provider_name,
+            function_name=function_name,
+            function_kwargs=function_kwargs,
+            locales=locales,
+        )
+
+        self._mapping_dict = {}
+        self._reverse_mapping_dict = {}
+        self._label_encoder = None
+
+    def get_mapping(self):
+        """Return the mapping dictionary."""
+        return deepcopy(self._mapping_dict)
+
+    def _fit(self, columns_data):
+        """Fit the transformer to the data.
+
+        Generate a ``_mapping_dict`` and a ``_reverse_mapping_dict`` for each
+        value in the provided ``columns_data`` using the ``Faker`` provider and
+        ``function``.
+
+        Args:
+            data (pandas.Series):
+                Data to fit the transformer to.
+        """
+        unique_values = columns_data[columns_data.notna()].unique()
+        unique_data_length = len(unique_values)
+        generated_values = [self._function() for _ in range(unique_data_length)]
+        if len(set(generated_values)) != unique_data_length:
+            counter = 0
+            while len(set(generated_values)) != unique_data_length:
+                generated_values.append(self._function())
+                counter += 1
+                if counter == self.MAX_ATTEMPTS_FOR_UNIQUE_VALUES:
+                    error_msg = (
+                        'The Faker function you specified is only able to generate '
+                        f'{len(set(generated_values))} unique values, which is not enough to '
+                        'create a mapping. Please use a different Faker function for this column.'
+                    )
+                    raise ValueError(error_msg)
```

Comment:
It looks like faker has a way to guarantee uniqueness https://faker.readthedocs.io/en/master/#unique-values. Have we tried this? Maybe we can remove the loop with this and say if it's unable to generate unique rows using this attribute we probably can't do it